qemu/tests/blk_commit.py

```diff
@@ -0,0 +1,99 @@
+import logging
+
+from autotest.client.shared import error
+
+from virttest import utils_misc
+
+from qemu.tests import block_copy
+
+
+class BlockCommit(block_copy.BlockCopy):
+
+    """
+    base class for block commit tests
+    """
+
+    def __init__(self, test, params, env, tag):
+        super(BlockCommit, self).__init__(test, params, env, tag)
+
+    @error.context_aware
+    def start(self):
+        """
+        start block device committing job;
+        """
+        base_image = self.params.get("base_image")
+        base_image = utils_misc.get_path(self.data_dir, base_image)
+        error.context("from start func: base image %s" % base_image)
+        top_image = self.params.get("top_image")
+        top_image = utils_misc.get_path(self.data_dir, top_image)
+        error.context("from start func: top image %s" % top_image)
+        default_speed = self.params.get("default_speed")
+
+        error.context("start to commit block device", logging.info)
+        self.vm.block_commit(self.device, default_speed, base_image, top_image)
+        status = self.get_status()
+        if not status:
+            raise error.TestFail("no active job found")
+        logging.info("block commit job running, with limited speed #{0}# B/s".format(default_speed))
+
+    @error.context_aware
+    def create_snapshots(self):
+        """
+        create live snapshot_chain, snapshots chain define in $snapshot_chain
+        """
+        image_format = self.params["snapshot_format"]
+        snapshots = self.params["snapshot_chain"].split()
+        error.context("create live snapshots %s" % snapshots, logging.info)
+        for snapshot in snapshots:
+            snapshot = utils_misc.get_path(self.data_dir, snapshot)
+            image_file = self.get_image_file()
+            device = self.vm.live_snapshot(image_file, snapshot, image_format)
+            if device != self.device:
+                image_file = self.get_image_file()
+                logging.info("expect file: #{0}#, opening file: #{1}#".format(snapshot, image_file))
+                raise error.TestFail("create snapshot '%s' fail" % snapshot)
+            self.trash_files.append(snapshot)
+
+    @error.context_aware
+    def job_finished(self):
```

Comment:
How about move this function to block_copy, we are not need to copy it again and again.